# HG changeset patch
# User Juan Pablo Carbajal <ajuanpi+dev@gmail.com>
# Date 1461266004 -7200
#      Thu Apr 21 21:13:24 2016 +0200
# Node ID 46ce9eb019a675769015ae21cba39ec5aeac849b
# Parent  bb4a05cb4a0257b9e1cf3e3ac368d072bd5cd4d3
Fix bug when tspan is a column vector (bug #47756).

* integrate_adaptive.m: Index into t, rather than tspan, so that orientation
does not matter.

diff --git a/scripts/ode/private/integrate_adaptive.m b/scripts/ode/private/integrate_adaptive.m
--- a/scripts/ode/private/integrate_adaptive.m
+++ b/scripts/ode/private/integrate_adaptive.m
@@ -146,17 +146,17 @@ function solution = integrate_adaptive (
                          & (dir * tspan(iout:end) <= dir * t_new));
         t_caught = t_caught + iout - 1;
 
         if (! isempty (t_caught))
           t(t_caught) = tspan(t_caught);
           iout = max (t_caught);
           x(:, t_caught) = ...
             runge_kutta_interpolate (order, [t_old t_new], [x_old x_new],
-                                     tspan(t_caught), new_k_vals, dt, func,
+                                     t(t_caught), new_k_vals, dt, func,
                                      options.funarguments);
 
           istep += 1;
 
           ## Call Events function only if a valid result has been found.
           ## Stop integration if eventbreak is true.
           if (options.haveeventfunction)
             break_loop = false;
